.phan/config.php

```diff
@@ -1,20 +1,38 @@
 <?php
 
 return [
-	"backward_compatibility_checks" => true,
 	// The docs on quick_mode at
-	// https://github.com/etsy/phan/wiki/Incrementally-Strengthening-Analysis
+	// https://github.com/phan/phan/wiki/Incrementally-Strengthening-Analysis
 	// don't seem reasonable. They claim that quick_mode=true add more errors.
 	// This is false on the assumption that that's a typo.
 	// It doesn't seem to have any effect on LORIS's codebase anyways.
 	"quick_mode" => false,
 	"analyze_signature_compatibility" => true,
+    // The default severity level for phan is 5. After removing all the
+    // suppressed rules, we should consider reducing this value to detect more
+    // suspicious code.
+    "minimum_severity" => 5,
+    // TODO: Update this value to the minimum required by LORIS.
+    "target_php_version" => "7.2",
 	// FIXME: allow_missing_properties should be false, but there's
 	// too many other things to fix first.
 	"allow_missing_properties" => true,
 	"null_casts_as_any_type" => false,
 	"scalar_implicit_cast" => false,
-	"ignore_undeclared_variables_in_global_scope" => false,
+    // The line below is required to prevent PhanUndeclaredVariable problems in
+    // the bvl_feedback ajax scripts. They all require() a file in this directory
+    // that declares the $feedbackThread variable. Phan doesn't know about this.
+    //
+    // Toggling the below rule to true is the suggested fix by the phan devs.
+    // see https://github.com/phan/phan/issues/1650.
+    //
+    // When the module is refactored, this line should be deleted.
+	"ignore_undeclared_variables_in_global_scope" => true,
+    // FIXME: We should add this.
+    "dead_code_detection" => false,
+    // FIXME: We should add this. Note that dead_code_detection also covers
+    // unused_variable_detection. It might be a good idea to enable this first.
+    "unused_variable_detection" => false,
```

Comment:
Are these finding things? phpmd already did/does these things, so we should have 0 results found.

Reply:
It's finding many things... I enabled just the `unused_variable_detection` rule and got 247 results after running phan.

PHPMD is also only configured to run on `php/libraries/`. 72 of the above results occurred in that folder so PHPMD is definitely missing something OR phan is giving a ton of false positives. Taking a look at the results though they seem accurate.

Reply:
(Some of the above errors are already fixed by my PRs that add new Phan rules.)